packages/amplify-provider-awscloudformation/lib/attach-backend.js

```diff
@@ -0,0 +1,299 @@
+  const currentAmplifyMeta = context.amplify.readJsonFile(currentAmplifyMetaFilePath);
+  if (!currentAmplifyMeta.providers[constants.ProviderName][constants.AmplifyAppIdLabel]) {
+    currentAmplifyMeta.providers[constants.ProviderName][constants.AmplifyAppIdLabel] = amplifyApp.appId;
+
+    const amplifyMetaFilePath = context.amplify.pathManager.getAmplifyMetaFilePath(process.cwd());
+    const jsonString = JSON.stringify(currentAmplifyMeta, null, 4);
+    fs.writeFileSync(currentAmplifyMetaFilePath, jsonString, 'utf8');
+    fs.writeFileSync(amplifyMetaFilePath, jsonString, 'utf8');
+
+    const { DeploymentBucketName } = currentAmplifyMeta.providers[constants.ProviderName];
+    await storeArtifactsForAmplifyService(context, awsConfig, DeploymentBucketName);
+  }
+
+  return currentAmplifyMeta;
+}
+
+async function storeArtifactsForAmplifyService(context, awsConfig, deploymentBucketName) {
+  const s3Client = getConfiguredS3Client(awsConfig);
+  const amplifyMetaFilePath = context.amplify.pathManager.getCurrentAmplifyMetaFilePath(process.cwd());
+  const backendConfigFilePath = context.amplify.pathManager.getCurrentBackendConfigFilePath(process.cwd());
+  await uploadFile(s3Client, deploymentBucketName, amplifyMetaFilePath);
+  await uploadFile(s3Client, deploymentBucketName, backendConfigFilePath);
+}
+
+async function uploadFile(s3Client, bucketName, filePath) {
+  if (!fs.existsSync(filePath)) {
+    return;
+  }
+  const key = path.basename(filePath);
+  const body = fs.createReadStream(filePath);
+  await s3Client
+    .putObject({
+      Bucket: bucketName,
+      Key: key,
+      Body: body,
+    })
+    .promise();
+}
+
+async function getAmplifyApp(context, amplifyClient) {
+  // If appId is in the inputParams, verify it
+  const { inputParams } = context.exeInfo;
+  if (inputParams.amplify && inputParams.amplify.appId) {
+    const inputAppId = inputParams.amplify.appId;
+    try {
+      const getAppResult = await amplifyClient
+        .getApp({
+          appId: inputAppId,
+        })
+        .promise();
+      context.print.info(`Input Amplify appId is verified: ${inputAppId}`);
+      context.print.info(`Amplify App name is: ${getAppResult.app.name}`);
+      return getAppResult.app;
+    } catch (e) {
+      context.print.error(`Input Amplify appId is invalild: ${inputAppId}`);
+      context.print.info(e);
+      throw e;
+    }
+  }
+
+  // If appId is not in the inputParams, prompt user to select
+  let apps = [];
+
+  let listAppsResponse = {};
+  do {
+    listAppsResponse = await amplifyClient
+      .listApps({
+        nextToken: listAppsResponse.nextToken,
+        maxResults: 25,
+      })
+      .promise();
+    apps = apps.concat(listAppsResponse.apps);
+  } while (listAppsResponse.nextToken);
+
+  if (apps.length > 1) {
+    const options = [];
+    apps.forEach(app => {
+      const option = {
+        name: `${app.name} / ${app.appId}`,
```

Comment:
In my feedback, I had just using ${app.name} and not ${app.appId}

Reply:
you got it